scripts/dev_setup.sh

```diff
@@ -45,83 +288,66 @@ else
 	exit 1
 fi
 
-cat <<EOF
-Welcome to Libra!
-
-This script will download and install the necessary dependencies needed to
-build Libra Core. This includes:
-	* Rust (and the necessary components, e.g. rust-fmt, clippy)
-	* CMake
-      * Clang
+if [[ $BATCH_MODE == "false" ]]; then
+    welcome_message
+    printf "Proceed with installing necessary dependencies? (y/N) > "
+    read -e input
+    if [[ "$input" != "y"* ]]; then
+	    echo "Exiting..."
+	    exit 0
+    fi
+fi
 
-If you'd prefer to install these dependencies yourself, please exit this script
-now with Ctrl-C.
+PRE_COMMAND=""
+if [ `whoami` != 'root' ]; then
+  PRE_COMMAND="sudo "
+fi
 
-EOF
+mkdir -p ${HOME}/bin
 
-printf "Proceed with installing necessary dependencies? (y/N) > "
-read -e input
-if [[ "$input" != "y"* ]]; then
-	echo "Exiting..."
-	exit 0
+if [[ $"$PACKAGE_MANAGER" == "apt-get" ]]; then
+	[[ $BATCH_MODE == "false" ]] && echo "Updating apt-get......"
+	$PRE_COMMAND apt-get update
 fi
 
-# Install Rust
-echo "Installing Rust......"
-if rustup --version &>/dev/null; then
-	echo "Rust is already installed"
-else
-	curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain stable
-	CARGO_ENV="$HOME/.cargo/env"
-	source "$CARGO_ENV"
-fi
+install_pkg cmake $PACKAGE_MANAGER
+install_pkg clang $PACKAGE_MANAGER
+install_pkg llvm $PACKAGE_MANAGER
+install_pkg curl $PACKAGE_MANAGER
+install_pkg gcc-powerpc-linux-gnu $PACKAGE_MANAGER
 
-# Run update in order to download and install the checked in toolchain
-rustup update
+#need to change....
+install_pkg libssl-dev $PACKAGE_MANAGER  # openssl-devel in centos
+install_pkg pkg-config $PACKAGE_MANAGER  # pkgconfig in centos
 
+install_rustup $BATCH_MODE
+[[ INSTALL_PROFILE == "false" ]] && install_cargo_profile
+install_toolchain `cat ./cargo-toolchain`
+install_toolchain `cat ./rust-toolchain`
 # Add all the components that we need
 rustup component add rustfmt
 rustup component add clippy
 
-if [[ $"$PACKAGE_MANAGER" == "apt-get" ]]; then
-	echo "Updating apt-get......"
-	sudo apt-get update
-fi
+install_sccache
+install_grcov
 
-echo "Installing CMake......"
-if which cmake &>/dev/null; then
-	echo "CMake is already installed"
-else
-	if [[ "$PACKAGE_MANAGER" == "yum" ]]; then
-		sudo yum install cmake -y
-	elif [[ "$PACKAGE_MANAGER" == "apt-get" ]]; then
-		sudo apt-get install cmake -y
-	elif [[ "$PACKAGE_MANAGER" == "pacman" ]]; then
-		sudo pacman -Syu cmake --noconfirm
-	elif [[ "$PACKAGE_MANAGER" == "brew" ]]; then
-		brew install cmake
-	fi
+if [[ $OPERATIONS == "true" ]]; then
+  install_pkg yamllint $PACKAGE_MANAGER
+  install_pkg python3 $PACKAGE_MANAGER
+  install_hadolint
+  install_vault
+  install_helm
+  install_terraform
+  install_kubectl
+  install_awscli
 fi
 
-echo "Installing Clang......"
```

Comment:
@metajack do you think we still need this?  I think most dev's laptop/workstation would have clang and llvm. This would cover those who don't.

Reply:
I'm still installing clang, just with the standard package manager tooling.
Definitely needed in CI.

Reply:
Oh, I see it now. They are moved up to lines 362 above.